### PR TITLE
arm-toolchain: add arm-toolchain/2021.07 recipe

### PR DESCRIPTION
**Pull request by madebr**
Add the toolchains from https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-a/downloads

Only target_os=Linux, target_arch=armv7hf/armv8 are currently added.
Also only for os=Linux, arch=x86_64

It's a WIP, so it needs more validation checks.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/arm-toolchain/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    settings = "os", "arch", "compiler"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _sources_dict(self):
+        try:
+            return self.conan_data["sources"][self.version][self._target_arch][self._target_os][str(self.settings.os)]
```
Comment (madebr) [inline recipes/arm-toolchain/all/conanfile.py:33]: TODO: check `self.settings.arch`

**Comment by SSE4** [comment]
is it GCC build?

**Comment by madebr** [comment]
> is it GCC build?

Yes, it's completely GNU.
I can't find a license in the tarball though.

**Comment by SSE4** [comment]
> Yes, it's completely GNU.
> I can't find a license in the tarball though.

but does it build from sources?

**Comment by madebr** [comment]
No, it's just re-packaging the binaries at the arm downloads page.

**Pull request closed without merging**